lib/paginator.rb

```diff
@@ -0,0 +1,61 @@
+class Paginator
+  PADDING = 2
+  SEPARATOR = "|"
+
+  attr_reader :current
+  attr_reader :total
+
+  def initialize(current, total)
+    @current = current.to_i || 1
+    @total = total
+  end
+
+  def pages
+    @pages ||= build_pages
```

Comment:
[suggestion] if we decided to go the Page object route, this could be a list of Pages!